extensions/common/context.cc

```diff
@@ -346,8 +346,9 @@ PeerNodeInfo::PeerNodeInfo(const std::string_view peer_metadata_id_key,
   // Attempt to read from filter_state first.
   found_ = getValue({peer_metadata_id_key}, &peer_id_);
   if (found_ && peer_id_ != kMetadataNotFoundValue) {
-    getValue({peer_metadata_key}, &peer_node_);
-    return;
+    if (getValue({peer_metadata_key}, &peer_node_)) {
```

Comment:
Should we set found_ to false?

Reply:
Sorry - went out to early. I wonder — should we update found to false when the second getValue fails?

Reply:
It's set to false two lines later

Reply:
Great, lgtm